v1/metrics/metrics.go

```diff
@@ -319,3 +327,38 @@ func Statistics(num ...int64) any {
 	}
 	return t.Value()
 }
+
+type noOpMetrics struct{}
+type noOpTimer struct{}
+type noOpHistogram struct{}
+type noOpCounter struct{}
+
+var (
+	noOpMetricsInstance   = &noOpMetrics{}
+	noOpTimerInstance     = &noOpTimer{}
+	noOpHistogramInstance = &noOpHistogram{}
+	noOpCounterInstance   = &noOpCounter{}
+)
+
+func (*noOpMetrics) Info() Info                      { return Info{Name: "<built-in no-op>"} }
+func (*noOpMetrics) Timer(name string) Timer         { return noOpTimerInstance }
```

Comment:
Technically, we could also do this

```suggestion
func (x *noOpMetrics) Timer(name string) Timer         { return x }
func (*noOpMetrics) Start()       {}
func (*noOpMetrics) Stop() int64  { return 0 }
func (*noOpMetrics) Value() any   { return 0 }
func (*noOpMetrics) Int64() int64 { return 0 }
```

etc...but it would make little difference.